src/core/ddsc/include/dds/ddsc/dds_public_qos.h

```diff
@@ -23,6 +23,11 @@
 #include "dds/export.h"
 #include "dds/ddsc/dds_public_qosdefs.h"
 
+/* Whether or not the "property list" QoS setting is supported in this version.  If it is,
+   the "dds.sec." properties are treated specially, preventing the accidental creation of
+   an non-secure participant by an implementation built without support for DDS Security. */
+#define DDS_HAS_PROPERTY_LIST_QOS 1
```

Comment:
Can you expose a `DDS_HAS_SECURITY_ENABLED` (or similar) as well? Otherwise we will need to assume security is enabled if `DDS_HAS_PROPERTY_LIST_QOS` is defined, which doesn't seem to be an accurate assumption to make unless I'm missing something.

Reply:
Yes, I can — and I have just pushed a commit that provides a ``DDS_HAS_SECURITY`` you can check at compile time.

It wasn't there at first because I thought it wasn't strictly necessary, because the non-security-enabled builds will (with this PR) outright refuse the creation of any DDS participant that requests security — and secondarily because it saved me from having to deal with CMake. But I think you're right that having it is indeed useful and the only downside (having to deal with CMake) is rather silly!

Reply:
> because it saved me from having to deal with CMake

Haha, been there man. Thanks for adding!